src/Microsoft.AspNetCore.WebUtilities/FileBufferingReadStream.cs

```diff
@@ -135,22 +180,27 @@ private Stream CreateTempFile()
         public override int Read(byte[] buffer, int offset, int count)
         {
             ThrowIfDisposed();
-            if (_buffer.Position < _buffer.Length || _completelyBuffered)
+            if (_buffer.Position < _readIntoBuffer || _completelyBuffered)
```

Comment:
MemoryStream.Length is not the length of the buffer, it's the length of the data. Contrast with Capacity which is the length of the buffer.